src/main/kotlin/com/openlattice/linking/controllers/RealtimeLinkingController.kt

```diff
@@ -52,6 +51,23 @@ class RealtimeLinkingController(
         return linkableEntitySets.minus(entitySetsNeedLinking)
     }
 
+    @RequestMapping(
+            path = [RealtimeLinkingApi.MISSING],
+            method = [RequestMethod.POST],
+            produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getEntitiesMissingLinking(@RequestBody entitySetIds: Set<UUID>): Map<UUID, Set<UUID>> {
```

Comment:
I would check, if the requested entity set ids are actually linkable entity sets and either throw `IllegalArgumentException` if we want to avoid calls like that or log a warning with the non-linkable entity set ids.
For that you will need to add hazelcastinstance to the constructor and get the entity sets map. I already did that for different reasons in this PR: https://github.com/openlattice/linker/pull/52/files You can copy those parts over.
To get the non-linkable entity set ids you could do something like this:
```
val linkableEntitySetIds = lqs
                .getLinkableEntitySets(linkableTypes, entitySetBlacklist, whitelist.orElse(setOf()))
                .toSet()
val nonLinkableEntitySetIds = entitySetIds.filter { !linkableEntitySets.contains(it) }
```
If you decide to throw error then:
```
require(nonLinkableEntitySetIds.isEmpty()) {"Some error message with $nonLinkableEntitySetIds"}
```

If not you can just log it:
```
if(nonLinkableEntitySetIds.isNotEmpty()) {
    logger.warn("Message with $nonLinkableEntitySetIds")
}
```

Reply:
An alternative could be to make a function in `LinkingQueryService` that returns the linkable entity set ids from requested entity set ids (and not all)

Reply:
Why not:
```
    override fun getEntitiesMissingLinking(@RequestBody entitySetIds: Set<UUID>): Map<UUID, Set<UUID>> {
        ensureAdminAccess()
        val linkableEntitySets = lqs
                .getLinkableEntitySets(linkableTypes, entitySetBlacklist, whitelist.orElse(setOf()))
                .toSet()
                .intersect(entitySetIds)
```
Taking the intersection between `linkableEntitySets` and `entitySetIds`?

Reply:
That's fine for getting the entitySetIds to pass to `lqs.getEntitiesNotLinked`, but I would still at least log entity set ids, that are non-linkable from the input ids